src/EPR.Calculator.Frontend/Controllers/LocalAuthorityDisposalCostsController.cs

```diff
@@ -1,78 +1,124 @@
+            this.clientFactory = clientFactory;
         }
 
-        private static List<LocalAuthorityViewModel> GetLocalAuthorityData()
+        /// <summary>
+        /// Handles the Index action asynchronously. Sends an HTTP request and processes the response to display local authority disposal costs.
+        /// </summary>
+        /// <returns>
+        /// An IActionResult that renders the LocalAuthorityDisposalCostsIndex view with grouped local authority data if the request is successful.
+        /// Redirects to the StandardErrorIndex action in case of an error.
+        /// </returns>
+        public IActionResult Index()
         {
-            var localAuthorityDisposalCosts = GetLocalAuthorityDisposalCosts();
+            try
+            {
+                Task<HttpResponseMessage> response = GetHttpRequest(this.configuration, this.clientFactory);
 
-            var localAuthorityData = new List<LocalAuthorityViewModel>();
+                if (response.Result.IsSuccessStatusCode)
+                {
+                    var deserializedlapcapdata = JsonConvert.DeserializeObject<List<LocalAuthorityDisposalCost>>(response.Result.Content.ReadAsStringAsync().Result);
 
-            if (localAuthorityDisposalCosts.Count > 0)
-            {
-                foreach (var la in localAuthorityDisposalCosts)
+                    // Ensure deserializedRuns is not null
+                    var localAuthorityDisposalCosts = deserializedlapcapdata ?? new List<LocalAuthorityDisposalCost>();
+                    var localAuthorityData = GetLocalAuthorityData(localAuthorityDisposalCosts);
+
+                    var localAuthorityDataGroupedByCountry = localAuthorityData?.GroupBy((data) => data.Country).ToList();
+
+                    return this.View(ViewNames.LocalAuthorityDisposalCostsIndex, localAuthorityDataGroupedByCountry);
+                }
+
+                if (response.Result.StatusCode == HttpStatusCode.NotFound)
                 {
-                    localAuthorityData.Add(new LocalAuthorityViewModel(la));
+                    return this.View();
                 }
+
+                return this.RedirectToAction(ActionNames.StandardErrorIndex, CommonUtil.GetControllerName(typeof(StandardErrorController)));
             }
+            catch (Exception)
+            {
+                return this.RedirectToAction(ActionNames.StandardErrorIndex, CommonUtil.GetControllerName(typeof(StandardErrorController)));
+            }
+        }
+
+        /// <summary>
+        /// Converts a list of LocalAuthorityDisposalCost objects into a list of LocalAuthorityViewModel objects.
+        /// </summary>
+        /// <param name="localAuthorityDisposalCosts">A list of LocalAuthorityDisposalCost objects to be converted.</param>
+        /// <returns>A list of LocalAuthorityViewModel objects.</returns>
+        private static List<LocalAuthorityViewModel> GetLocalAuthorityData(List<LocalAuthorityDisposalCost> localAuthorityDisposalCosts)
```

Comment:
Can we have some unit tests for `GetLocalAuthorityData`

Reply:
Also, please move it to Util class so that it can be reused if needed. Thanks

Reply:
sure

Reply:
done that